packages/shared/src/variants/fractional/fractional.test.ts

```diff
@@ -65,3 +60,27 @@ test("Surrounded merge stone", () => {
   expect(state.boardState.filter((i) => i).length).toBe(2);
   expect(state.boardState.at(Number(cornerId))).toBeNull();
 });
+
+test("A surrounded stone with other long chains on the board", () => {
+  const game = new FractionalTestGame({
+    players: [
+      { primaryColor: "black", secondaryColor: "green" },
+      { primaryColor: "white", secondaryColor: "green" },
+    ],
+  });
+
+  expect(game.corners).toHaveLength(4);
+
+  game.playMove(0, game.idOf(game.corners[3]));
+  game.playMove(1, game.idOf(game.corners[3].neighbours[0]));
+
+  for (let i = 8; i >= 6; --i) {
```

Comment:
Might be good to comment what is important about these moves. It's a little hard to parse out since a reader needs to do index calculation and know this is 9x9 board.  Or even a little ASCII art:

```
// . . . . . B B B B  // Additional chains on the board
// . . . . . . W W W  // have historically caused issues
// . ...5 lines... .
// W . . . . . . . .
//[B]W . . . . . . .  // White's last move is to capture black
```